File: 백준/Silver/1439. 뒤집기/뒤집기.py
one, zero = 0, 0

letter = input()

if letter[0] == '1':
  one += 1
else:
  zero += 1

for i in range(1, len(letter)):
  if letter[i - 1] != letter[i]:
    if letter[i] == '1' :
      one += 1
    else:
      zero += 1
      
print(min(one, zero))

File: 백준/Silver/1439. 뒤집기/뒤집기.c
#include <stdio.h>
#include <string.h>

int main(void) { 
   int one = 0, zero = 0;
   char str[1000001];

   scanf("%s", str);

   if(str[0] == '1') one++;
   else zero++;
   
   for(int i = 1; i < strlen(str); i++){
      if(str[i - 1] != str[i]){
         if(str[i] == '1') one++;
         else zero++;
      }
   }

   if(zero < one) printf("%d", zero);
   else printf("%d", one);
   
   return 0;
}